rev'd up to 1.4.0

peter-kwan opened the pull request.

EvanLeleux commented on praetorian_cli/sdk/chariot.py:
it looks like we are allowing headers to be passed in here to then be reassigned. maybe just change the request() method to merge headers into kwargs['headers'] then remove the headers=headers? Dont know if that works

peter-kwan replied:
Good idea. I read your comment this way -- chariot_request() and request() are sort of redundant. In the original code, request() takes care of proxy. and chariot_request() takes care of the beta flag and adding the authentication headers.

Why not just have request() does all of it. 

And this would simplify the handling of headers.

In fact, not one single original callers of chariot_request() sets the headers argument.

I will simplify it.

peter-kwan replied:
addressed here: de21d854

peter-kwan commented:
@EvanLeleux  this is ready to be reviewed again. I addressed your comment to simplify chariot_request() and request() in chariot.py. 

This also fixes a bunch of test failures that I got running the tests on `main`.

peter-kwan commented:
Also ready to be published next time around since the backend in prod supports the _encrypted/ partition now.
